File: data/derived_cot/rq2_eval/translation/qwen/_converted_code/java_to_cpp/Warehouse.cpp
#include <unordered_map>
#include <string>

class Warehouse {
private:
    struct Product {
        std::string name;
        int quantity;

        Product(std::string name, int quantity) : name(name), quantity(quantity) {}
        void addQuantity(int quantity) { this->quantity += quantity; }
        int getQuantity() const { return quantity; }
    };

    struct Order {
        int productId;
        int quantity;
        std::string status;

        Order(int productId, int quantity, std::string status) 
            : productId(productId), quantity(quantity), status(std::move(status)) {}
        void setStatus(std::string status) { this->status = std::move(status); }
        std::string getStatus() const { return status; }
    };

    std::unordered_map<int, Product> inventory;
    std::unordered_map<int, Order> orders;

public:
    Warehouse() = default;

    void add_product(int productId, std::string name, int quantity) {
        if (inventory.find(productId) != inventory.end()) {
            inventory[productId].addQuantity(quantity);
        } else {
            inventory[productId] = Product(name, quantity);
        }
    }

    void update_product_quantity(int productId, int quantity) {
        if (inventory.find(productId) != inventory.end()) {
            inventory[productId].addQuantity(quantity);
        }
    }

    int get_product_quantity(int productId) const {
        if (inventory.find(productId) != inventory.end()) {
            return inventory[productId].getQuantity();
        }
        return -1;
    }

    bool create_order(int orderId, int productId, int quantity) {
        if (inventory.find(productId) != inventory.end() && 
            inventory[productId].getQuantity() >= quantity) {
            inventory[productId].addQuantity(-quantity);
            orders[orderId] = Order(productId, quantity, "Shipped");
            return true;
        }
        return false;
    }

    bool change_order_status(int orderId, std::string status) {
        if (orders.find(orderId) != orders.end()) {
            orders[orderId].setStatus(status);
            return true;
        }
        return false;
    }

    std::string track_order(int orderId) const {
        if (orders.find(orderId) != orders.end()) {
            return orders[orderId].getStatus();
        }
        return "";
    }
};

File: data/derived_cot/rq2_eval/translation/r1/_converted_code/java_to_py/Warehouse.py
class Warehouse:

    class _Product:

        def __init__(self, name, quantity):
            self.name = name
            self.quantity = quantity

        def addQuantity(self, quantity):
            self.quantity += quantity

        def getQuantity(self):
            return self.quantity

    class _Order:

        def __init__(self, productId, quantity, status):
            self.productId = productId
            self.quantity = quantity
            self.status = status

        def getStatus(self):
            return self.status

        def setStatus(self, status):
            self.status = status

    def __init__(self):
        self.inventory = {}
        self.orders = {}

    def add_product(self, productId, name, quantity):
        if productId in self.inventory:
            self.inventory[productId].addQuantity(quantity)
        else:
            self.inventory[productId] = self._Product(name, quantity)

    def update_product_quantity(self, productId, quantity):
        if productId in self.inventory:
            self.inventory[productId].addQuantity(quantity)

    def get_product_quantity(self, productId):
        if productId in self.inventory:
            return self.inventory[productId].getQuantity()
        else:
            return -1

    def create_order(self, orderId, productId, quantity):
        if productId in self.inventory and self.inventory[productId].getQuantity() >= quantity:
            self.inventory[productId].addQuantity(-quantity)
            self.orders[orderId] = self._Order(productId, quantity, 'Shipped')
            return True
        else:
            return False

    def change_order_status(self, orderId, status):
        if orderId in self.orders:
            self.orders[orderId].setStatus(status)
            return True
        else:
            return False

    def track_order(self, orderId):
        if orderId in self.orders:
            return self.orders[orderId].getStatus()
        else:
            return None